File: 8kyu/(8 kyu) You Cant Code Under Pressure 1/(8 kyu) You Cant Code Under Pressure 1.dart
// 1 Plain solution
int doubleInteger(int i) {
  return i * 2;
}

// 2 Optimized solution
int doubleInteger(int i) {
  return i + i;
}

// 3 Clever solution
int doubleInteger(int i) {
  return i << 1;
}

// 4 Coding golf
int doubleInteger(int i) => i << 1;

File: 8kyu/(8 kyu) You Cant Code Under Pressure 1/(8 kyu) You Cant Code Under Pressure 1.ts
// 1 Plain solution
export function doubleInteger(i: number): number {
  return 2 * i;
}

// 2 Optimized solution
export const doubleInteger = (i: number): number => 2 * i;

// 3 Coding golf
export const doubleInteger = (i: any) => i << 1;
